detection_rules/ecs.py

```diff
@@ -259,12 +269,48 @@ def download_schemas(refresh_master=True, refresh_all=False, verbose=True):
             shutil.rmtree(m, ignore_errors=True)
 
         master_dir = "master_{}".format(master_ver)
-        os.makedirs(get_etc_path(ETC_NAME, master_dir), exist_ok=True)
+        os.makedirs(get_etc_path(ECS_NAME, master_dir), exist_ok=True)
 
         compressed = gzip_compress(json.dumps(master_schema, sort_keys=True, cls=DateTimeEncoder))
-        new_path = get_etc_path(ETC_NAME, master_dir, "ecs_flat.json.gz")
+        new_path = get_etc_path(ECS_NAME, master_dir, "ecs_flat.json.gz")
         with open(new_path, 'wb') as f:
             f.write(compressed)
 
         if verbose:
             print('Saved files to {}: \n\t- {}'.format(master_dir, 'ecs_flat.json.gz'))
+
+
+def download_endpoint_schemas(target: str, overwrite: bool = True) -> None:
+    """Download endpoint custom schemas."""
+    url = "https://raw.githubusercontent.com/elastic/endpoint-package/main/custom_schemas"
+    r = requests.get(f"{url}/custom_{target}.yml")
+    r.raise_for_status()
+    schema = yaml.safe_load(r.text)[0]
+    root_name = schema["name"]
+    fields = schema["fields"]
+    flattened = {}
+
+    for f in fields:
+        if 'multi_fields' in f:
+            for mf in f['multi_fields']:
+                flattened[f"{root_name}.{f['name']}.{mf['name']}"] = mf['type']
+        else:
+            flattened[f"{root_name}.{f['name']}"] = f['type']
+
+    Path(ENDPOINT_SCHEMAS_DIR).mkdir(parents=True, exist_ok=True)
+    compressed = gzip_compress(json.dumps(flattened, sort_keys=True, cls=DateTimeEncoder))
+    new_path = Path(ENDPOINT_SCHEMAS_DIR) / f"endpoint_{target}.json.gz"
+    if overwrite:
+        shutil.rmtree(new_path, ignore_errors=True)
+    with open(new_path, 'wb') as f:
```

Comment:
FWIW, I would check `download_endpoint_package_schema` in ER which has similar functionality. I believe @brokensound77 mentioned a CI job to migrate those schemas over to DR every time we built there since we're already doing that.

Reply:
Confirmed with @Mikaayenson that Endpoint Rules pulls in the tagged releases of Endpoint Package, but does not include the fields available in the custom schemas. Therefore this is introducing a new schema to our rule repositories.